utils/gen_onnx_mlir.py

```diff
@@ -425,6 +425,29 @@
    auto resultType = mlir::UnrankedTensorType::get(to.getValue());
    build($_builder, $_state, resultType, input, to);
   }] >
+  ];'''),
+  ('Split',
+ '''   let builders = [
+  OpBuilder<(ins "Value":$input, "IntegerAttr":$axis, "ArrayAttr":$split), [{
+   Type elementType = input.getType().cast<ShapedType>().getElementType();
+   std::vector<mlir::Type> outputTypes;
+   if (split.size() == 0) {
+     outputTypes.emplace_back(UnrankedTensorType::get(elementType));
+   } else {
+     std::vector<int64_t> inputShape = input.getType().cast<MemRefType>().getShape();
```

Comment:
How is the cast<MemRefType>() guaranteed to be successful? For MemRefType, do we need to worry unknown dimension?

Reply:
Its shapeInference code guarantees that the input tensor is RankedTensorType. I fixed the code to convert the input tensor to UnrankedTensorType.